lib/hammox.ex

```diff
@@ -109,10 +109,25 @@ defmodule Hammox do
 
   add_2.(1.5, 2.5) # throws Hammox.TypeMatchError
   ```
+
+  If pass module and list instead of MFA and module works like `protect/3`
+  ```elixir
+  # calling this
+  Hammox.protect(SomeModule, foo: 1)
+  # works same as this
+  Hammox.protect(SomeModule, SomeModule, foo: 1)
```

Comment:
I'd explain the rationale behind this to be even more helpful!
```suggestion
  A common tactic is to put behaviour callbacks and the "default"
  implementations for these callbacks in the same module. For these kinds of
  modules, you can also use `protect/2` as a shortcut for `protect/3`:
  ```elixir
  # calling this
  Hammox.protect(SomeModule, foo: 1)
  # works the same as this
  Hammox.protect(SomeModule, SomeModule, foo: 1)
  ```
```